src/node.js

```diff
@@ -91,6 +91,15 @@
         var path = NativeModule.require('path');
         process.argv[1] = path.resolve(process.argv[1]);
 
+        // check if user passed `-c` or `--check` arguments to Node.
+        if (process._syntax_check_only != null) {
+            var vm = NativeModule.require('vm');
+            var fs = NativeModule.require('fs');
+            var source = fs.readFileSync(process.argv[1], 'utf-8').replace(/^#.*/, '');
```

Comment:
I see a couple of issues here:

1. It won't work for filenames without an extension (`node app` vs. `node app.js`)
2. It won't skip the UTF-8 BOM (byte order marker) like `require()` does.
3. The regular expression doesn't match the one in lib/module.js that matches the string `#!`.

It's probably unavoidable that fixing 2 and 3 results in some duplicated code but it should come with regression tests that ensure the implementations won't diverge.